Makefile

```diff
@@ -82,57 +89,118 @@ for-all-target: $(ALL_MODS)
 integration-vet:
 	@set -e; cd tests && go vet -tags integration,testutilsintegration,zeroconfig,testutils ./... && $(GOTEST_SERIAL) $(BUILD_INFO_TESTS) -tags testutils,testutilsintegration -v -timeout 5m -count 1 ./...
 
+.PHONY: integration-test-target
+integration-test-target:
+	@set -e; cd tests && $(GOTEST_SERIAL) $(BUILD_INFO_TESTS) --tags=$(TARGET) -v -timeout 5m -count 1 ./...
+
+.PHONY: integration-test-cover-target
+integration-test-cover-target:
+	@set -e; $(MAKE_TEST_COVER_DIR) && cd tests && $(GOTEST_SERIAL) $(BUILD_INFO_TESTS) --tags=$(TARGET) -v -timeout 5m -count 1 ./... $(COVER_TESTING_INTEGRATION_OPTS)
+	$(GOCMD) tool covdata textfmt -i=$(TEST_COVER_DIR) -o ./$(TARGET)-coverage.txt
+
 .PHONY: integration-test
 integration-test:
-	@set -e; cd tests && $(GOTEST_SERIAL) $(BUILD_INFO_TESTS) --tags=integration -v -timeout 5m -count 1 ./...
+	@make integration-test-target TARGET='integration'
+
+.PHONY: integration-test-with-cover
+integration-test-with-cover:
+	@make integration-test-cover-target TARGET='integration'
 
 .PHONY: integration-test-mongodb-discovery
 integration-test-mongodb-discovery:
-	@set -e; cd tests && $(GOTEST_SERIAL) $(BUILD_INFO_TESTS) --tags=discovery_integration_mongodb -v -timeout 5m -count 1 ./...
+	@make integration-test TARGET='discovery_integration_mongodb'
+
+.PHONY: integration-test-mongodb-discovery-with-cover
```

Comment:
I've decided to introduce a second target for each integration test with `-with-cover` appended to the target name to denote that code coverage is enabled for the test. The alternative approach would be to use a single target and passing a boolean to say whether cover testing is wanted or not. I think passing a boolean is less clear for the user, and more prone to mistakes. The approach of two separate targets does make the `Makefile` a decent bit bigger, but I think the UX is more clear this way.